beacon-chain/rpc/beacon/blocks.go

```diff
@@ -148,20 +149,61 @@ func (bs *Server) ListBlocks(
 func (bs *Server) GetChainHead(ctx context.Context, _ *ptypes.Empty) (*ethpb.ChainHead, error) {
 	headState, err := bs.HeadFetcher.HeadState(ctx)
 	if err != nil {
-		return nil, status.Errorf(codes.InvalidArgument, "Could not get head state")
+		return nil, status.Errorf(codes.Internal, "Could not get head state: %v", err)
 	}
+
+	headBlock := bs.HeadFetcher.HeadBlock()
+	headBlockRoot, err := ssz.SigningRoot(headBlock)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not get head block root: %v", err)
+	}
+
 	finalizedCheckpoint := headState.FinalizedCheckpoint
+	finalizedBlockSlot, err := bs.blockSlot(ctx, finalizedCheckpoint.Epoch*params.BeaconConfig().SlotsPerEpoch)
```

Comment:
Why not look up the block by hash and read it's slot value? You have `finalizedCheckpoint.Root`.